tests/subsys/pm/device_runtime/src/main.c

```diff
@@ -0,0 +1,240 @@
+/*
+ * Copyright (c) 2021 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <sys/printk.h>
+#include <zephyr.h>
+#include <zephyr/types.h>
+#include <ztest.h>
+#include <ksched.h>
+#include <kernel.h>
+#include <pm/pm.h>
+#include <pm/device_runtime.h>
+#include "dummy_driver.h"
+
+#define MAX_TIMES 10
+#define STACKSIZE 1024
+
+/* Semaphore used to synchronize thread A and thread B*/
+K_SEM_DEFINE(sem, 0, 1);
+
+K_THREAD_STACK_DEFINE(threadA_stack, STACKSIZE);
+K_THREAD_STACK_DEFINE(threadB_stack, STACKSIZE);
+
+static const struct device *dev;
+static struct dummy_driver_api *api;
+
+static struct k_thread threadA;
+static struct k_thread threadB;
+
+
+void threadA_func(void *arg1, void *arg2, void *arg3)
+{
+	int ret;
+
+	ARG_UNUSED(arg1);
+	ARG_UNUSED(arg2);
+	ARG_UNUSED(arg3);
+
+	ret = pm_device_get(dev);
```

Comment:
why is the thread calling this directly? Shouldn't you call driver's dummy_open()?

Reply:
It could be, but then I would need to call pm_device_get_sync() to test the synchronous path ... To avoid further mistakes, I've just added open_sync and close_sync to the driver and calling them instead of using directly the get/put.

Reply:
well, create another driver function that uses get_sync to test this path, I think this will be confusing and error prone if we start calling those functions outside of a driver and we will have no way of testing the real use case.